### PR TITLE
Global shell settings

### PR DESCRIPTION
**Pull request by roboguy13**
Switched this over to use `IORef`s. (https://github.com/ku-fpg/hermit-shell/issues/29)

### CODE REVIEW

File: src/HERMIT/API/Types.hs [hunk 1]
```diff
@@ -6,6 +6,8 @@
 {-# LANGUAGE DeriveDataTypeable #-}
 {-# LANGUAGE TypeSynonymInstances #-}
 {-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
```
Comment (RyanGlScott) [inline src/HERMIT/API/Types.hs]: This isn't needed anymore, right?

File: src/HERMIT/API/Types.hs [hunk 2]
```diff
@@ -89,15 +93,27 @@ proxyToJSON Proxy = String $ pack $ show $ typeOf (undefined :: a)
...
 instance Response () where
   printResponse () = return ()
 
 instance Response String where
-  printResponse = print
+  printResponse str = do
+    quietMode <- readIORef quietModeM
+    if quietMode
+      then return ()
+      else print str
```
Comment (RyanGlScott) [inline src/HERMIT/API/Types.hs]: Consider refactoring lines like this into `when quietMode $ print str`. (Also, convert occurrences of `when debug` to `when quietMode`.)

**Comment by roboguy13** [comment]
@RyanGlScott Good points! Should debug output be controlled by the quiet mode option? I was thinking that quiet mode would be if we don't want to print out every step of a script (which gives us behavior more like HERMIT's original behavior).

**Comment by RyanGlScott** [comment]
If you want to separate shell output from developer-specific output, then ignore what I said about `when debug`. :)

**Pull request merged**